### PR TITLE
feat: add `icon-button` component

### PR DESCRIPTION
**Pull request by mseyfayi**

### CODE REVIEW

**Comment by majidsajadi** [comment]
I think it's not a good idea to add a property to differentiate a normal button from the icon button. I think its better to add a `tap-icon-button` element to support this functionality.

**Comment by mseyfayi** [comment]
> I think it's not a good idea to add a property to differentiate a normal button from the icon button. I think its better to add a `tap-icon-button` element to support this functionality.

BaseButton created, Button & IconButton are extending it and using its styles

File: src/base-button/index.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+@customElement("tap-base-button")
+export class TapBaseButton extends BaseButton {
+  static readonly styles = [styles];
+}
+
+declare global {
+  interface HTMLElementTagNameMap {
+    "tap-base-button": TapBaseButton;
+  }
+}
```
Comment (majidsajadi) [inline src/base-button/index.ts:14]: we do not have to define a custom element `tap-base-button`. the base class is enough.

File: src/base-button/base-button.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    | "elevated"
+    | "destructive"
+    | "brand" = "primary";
+
+  constructor() {
+    super();
+    this.internals = this.attachInternals();
+  }
+
+  protected handleSlotChange = () => {};
```
Comment (majidsajadi) [inline src/base-button/base-button.ts:42]: please remove this handle if we dont need it.

File: src/base-button/base-button.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        id="button"
+        class="button"
+        role="button"
+        part="button"
+        @click=${this.handleClick}
+        ?disabled="${this.disabled}"
+        type="${ifDefined(this.type)}"
+        name=${ifDefined(this.name)}
+        value=${ifDefined(this.value)}
+        aria-label="${ifDefined(this.label)}"
```
Comment (majidsajadi) [inline src/base-button/base-button.ts:66]: please remove the quotations.

File: src/icon-button/icon-button.style.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { css } from "lit";
+
+export default css`
+  :host {
+    border-radius: var(--tap-sys-radius-full);
+  }
+
+  :host([size="small"])::part(button) {
```
Comment (majidsajadi) [inline src/icon-button/icon-button.style.ts:8]: part is not well supported in the browsers.

File: src/base-button/base-button.style.ts [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+    background: var(
+      --tap-button-color-gradient-brand,
+      var(--tap-sys-color-gradient-brand)
+    );
+    color: var(
+      --tap-button-color-content-on-inverse,
+      var(--tap-sys-color-content-on-inverse)
+    );
+  }
+`;
```
Comment (majidsajadi) [inline src/base-button/base-button.style.ts:195]: please only keep the shared styles in the base button styles. and extract the `button` styles to its own style file. for example size related styles.

**Pull request merged**